workflow/src/test/kotlin/com/r3/corda/sdk/token/workflow/statesAndContracts/TradeConditions.kt

```diff
@@ -0,0 +1,90 @@
+package com.r3.corda.sdk.token.workflow.statesAndContracts
+
+import net.corda.core.contracts.*
+import net.corda.core.identity.AbstractParty
+import net.corda.core.internal.castIfPossible
+import net.corda.core.schemas.MappedSchema
+import net.corda.core.schemas.PersistentState
+import net.corda.core.schemas.QueryableState
+import net.corda.core.serialization.CordaSerializable
+import net.corda.core.transactions.LedgerTransaction
+import java.util.*
+import javax.persistence.Column
+import javax.persistence.Entity
+import javax.persistence.Table
+
+@BelongsToContract(TradeConditionsContract::class)
+data class TradeConditions(val owner: AbstractParty,
```

Comment:
Instead of `TradeConditions` being a simple string, how about converting it to an encumbrance? This way you can model how an actual state constraint would work in Corda. (Of course, be aware of the caveats of not mixing an encumbrance with a reference state; this will work fine for tokens with a fixed (inline) definition but not for tokens using an EvolvableToken token type.)